core/src/main/java/com/graphhopper/storage/CHProfile.java

```diff
@@ -20,14 +20,18 @@
 public class CHProfile {
     private final Weighting weighting;
     private final boolean edgeBased;
-    private final int uTurnCosts;
 
     public static CHProfile nodeBased(Weighting weighting) {
-        return new CHProfile(weighting, TraversalMode.NODE_BASED, INFINITE_U_TURN_COSTS);
+        // todonow: should we do such a check (check if the weighting has turn costs ?)
+//        if (!edgeBased && uTurnCosts != INFINITE_U_TURN_COSTS) {
+//            throw new IllegalArgumentException("Finite u-turn costs are only allowed for edge-based CH");
+//        }
+        return new CHProfile(weighting, TraversalMode.NODE_BASED);
     }
 
-    public static CHProfile edgeBased(Weighting weighting, int uTurnCosts) {
-        return new CHProfile(weighting, TraversalMode.EDGE_BASED, uTurnCosts);
+    public static CHProfile edgeBased(Weighting weighting) {
+        // todonow: should we check/assure that the weighting must have some valid turn costs ?
```

Comment:
Similar question here. But maybe we even want edge-based without turn costs for some(?) reason. #1702

Reply:
I removed the check, here I actually think its bad to do it because of #1702.

Reply:
Ok